File: clients/java1.8_client/client/src/main/java/core/API/Elevator.java
package core.API;

import org.json.simple.JSONArray;
import org.json.simple.JSONObject;

import java.util.*;

/**
 * Created by Boris on 01.02.17.
 */
public class Elevator implements MessagesInterface {
    private Integer id;
    private Double y;
    private List<Passenger> passengers;
    private Integer state;
    private Double speed;
    private Integer timeOnFloor;
    private Integer floor;
    private String type;
    private Integer nextFloor;
    private List<JSONObject> messages;

    public Integer getState() {
        return this.state;
    }

    public Double getSpeed() {
        return this.speed;
    }

    public Integer getNextFloor() { return this.nextFloor; }

    public Integer getTimeOnFloor() {
        return this.timeOnFloor;
    }

    public Integer getFloor() {
        return this.floor;
    }

    public String getType() {
        return this.type;
    }

    public Double getY() {
        return this.y;
    }

    public List<Passenger> getPassengers() {
        return this.passengers;
    }

    public Integer getId() {
        return this.id;
    }

    public List<JSONObject> getMessages(){
        return this.messages;
    }

    public Elevator(JSONObject elevator) {
        id = (int) (long) elevator.get("id");
        if (elevator.get("y") instanceof Long) {
            y = ((Long) elevator.get("y")).doubleValue();
        } else {
            y = (double) elevator.get("y");
        }
        passengers = new ArrayList<>();
        for (Object passenger : (JSONArray) elevator.get("passengers")) {
            passengers.add(new Passenger((JSONObject) passenger));
        }
        state = (int) (long) elevator.get("state");
        speed = (double) elevator.get("speed");
        timeOnFloor = (int) (long) elevator.get("time_on_floor");
        floor = (int) (long) elevator.get("floor");
        type = (String) elevator.get("type");
        nextFloor = (int) (long) elevator.get("next_floor");
        messages = new ArrayList<>();
    }

    public void goToFloor(Integer floor) {
        this.nextFloor = floor;
        JSONObject jo = new JSONObject();
        jo.put("command", "go_to_floor");
        JSONObject args = new JSONObject();
        args.put("elevator_id", this.id);
        args.put("floor", floor);
        jo.put("args", args);
        this.messages.add(jo);
    }
}

File: clients/kotlin_client/client/src/main/kotlin/core/API/Elevator.kt
package core.API

import org.json.simple.JSONArray
import org.json.simple.JSONObject

class Elevator(elevator: JSONObject) : MessagesInterface {
  val id: Int = elevator["id"]?.toString()?.toInt() ?: 0
  val y: Double = elevator["y"]?.toString()?.toDouble() ?: 0.0
  val passengers = ArrayList<Passenger>()
  val state: Int = elevator["state"]?.toString()?.toInt() ?: 0
  val speed: Double = elevator["speed"]?.toString()?.toDouble() ?: 0.0
  val timeOnFloor: Int = elevator["time_on_floor"]?.toString()?.toInt() ?: 0
  val floor: Int = elevator["floor"]?.toString()?.toInt() ?: 0
  val type: String = elevator["type"]?.toString() ?: ""
  var nextFloor: Int = elevator["next_floor"]?.toString()?.toInt() ?: 0
    private set
  val messages = ArrayList<JSONObject>()

  init {
    (elevator["passengers"] as JSONArray).mapTo(passengers) { Passenger(it as JSONObject) }
  }

  override fun getMessages(): List<JSONObject> = messages

  fun goToFloor(floor: Int?) {
    this.nextFloor = floor ?: 0
    val jo = JSONObject()
    jo.put("command", "go_to_floor")
    val args = JSONObject()
    args.put("elevator_id", this.id)
    args.put("floor", floor)
    jo.put("args", args)
    this.messages.add(jo)
  }
}